src/jvmMain/kotlin/io/beatmaps/api/search.kt

```diff
@@ -197,6 +200,16 @@ fun Route.searchRoute() {
         val sortArgs = searchInfo.sortArgsFor(actualSortOrder)
 
         newSuspendedTransaction {
+            val followingSubQuery = if (it.followed == true) {
+                requireAuthorization { _, user ->
```

Comment:
This shouldn't be a call to requireAuthorization, it very much seems that when not logged in you just ignore the property.
The call has the side effect of returning an error to the user when not logged in.

Just wrap the whole handler in optionalAuthorization and check if the user is null / auth type is None

Reply:
(Which will mean you won't need those changes to `requireAuthorization` and `optionalAuthorization`)